utils/obsproc/Smap2Ioda.h

```diff
@@ -83,21 +85,47 @@ namespace gdasapp {
 
       int year = startYear;
       int month = 1;
-      int day = 1;
+      int day = startDay;
+      while (day > 31) {
+          if (month == 4 || month == 6 || month == 9 || month == 11) {
+              day -= 30;
+          } else if (month == 2) {
+              day -= 28;  // Assuming not a leap year
+          } else {
+              day -= 31;
+          }
+          month++;
+      }
 
       // Replace Fillvalue -9999 to 0 to avoid crash in dateToJulian
       if (year == -9999) {
-        year = 0;
+        year = month = day = 0;
       }
 
-      // Convert a date to Julian date
-      uint64_t julianDate = util::datefunctions::dateToJulian(year, month, day);
+      // Set hour, minute, and second to 0
+      int hour = 0, minute = 0, second = 0;
+
+      // Construct iso8601 string format for each dateTime
+      std::stringstream ss;
+      ss << std::setfill('0')
+         << std::setw(4) << year << '-'
+         << std::setw(2) << month << '-'
+         << std::setw(2) << day << 'T'
+         << std::setw(2) << hour << ':'
+         << std::setw(2) << minute << ':'
+         << std::setw(2) << second << 'Z';
+      std::string formattedDateTime = ss.str();
+
+      // Create util::DateTime object from formatted string
+      util::DateTime dateTime(formattedDateTime);
 
-      // Subtract Julian day from January 1, 1970 (convert to epoch)
-      int daysSinceEpoch = julianDate - 2440588;
+      // Set epoch time for SMAP_SSS
+      util::DateTime epochDtime("1970-01-01T00:00:00Z");
 
-      // Calculate seconds
-      int secondsSinceEpoch = (daysSinceEpoch + startDay) * 86400;
+      // Convert Obs DateTime objects to epoch time offsets in seconds
+      // 0000-00-00T00:00:00Z will be converterd to negative seconds
+      int64_t timeOffsets
+           = ioda::convertDtimeToTimeOffsets(epochDtime, {dateTime})[0];
 
```

Comment:
we shouldn't have to do this.
Try to make use of boost:
```c++
#include <boost/date_time/gregorian/gregorian.hpp>
```
has a bunch of nifty datetime utilities that probably would have saved us some headache.